gcpy/cstools.py

```diff
@@ -692,10 +689,113 @@ def is_cubed_sphere(
     is_gchp : bool
         Returns True if data is placed on a cubed-sphere grid,
         and False otherwise.
+
+    Remarks:
+    --------
+    A cubed-sphere data file has one of the following attributes
+    (1) A dimension named "nf" (GCHP/GEOS diagnostic files)
+    (2) The lat/lon ratio is exactly 6 (GCHP/GEOS checkpoints)
+    """
+    gcpy.util.verify_variable_type(data, (xr.DataArray, xr.Dataset))
+
+    if is_cubed_sphere_diag_grid(data):
+        return True
+    if is_cubed_sphere_rst_grid(data):
+        return True
+    return False
+
+
+def is_cubed_sphere_diag_grid(data):
+    """
+    Determines if a cubed-sphere grid has History file dimensions.
+    (i.e. a dimension named "nf", ak number of grid faces).
+
+    Args:
+    -----
+    data : xarray.DataArray or xarray.Dataset
+        The input data.
+
+    Returns:
+    --------
+    True if the grid has restart dimensions, False otherwise.
+    """
+    if "nf" in data.dims:
+        return True
+    return False
+
+
+def is_cubed_sphere_rst_grid(data):
+    """
+    Determines if a cubed-sphere grid has restart file dimensions.
+    (i.e. lat and lon, with lat = lon*6).
+
+    Args:
+    -----
+    data : xarray.DataArray or xarray.Dataset
+        The input data.
+
+    Returns:
+    --------
+    True if the grid has restart dimensions, False otherwise.
     """
     gcpy.util.verify_variable_type(data, (xr.DataArray, xr.Dataset))
 
-    if "nf" in data.dims:   # nf = number of cubed-sphere faces
+    if "lat" in data.dims:
+        if data.dims["lat"] == data.dims["lon"] * 6:
+            return True
+    if "SPC_" in data.data_vars.keys():
```

Comment:
Why is this extra check here? Are there cases of a CS restart file where lat is not lon*6?

Reply:
Thanks @lizziel.  My thinking was if this function gets passed a lat-lon grid then you would need the extra check or else it might flag a GCClassic file as a GCHP restart file.  Then it would just fall through and return false.  Also probably the first check is faster than looking for "SPC_" in the data variables list.  That was my thinking anyway.